optimum/quanto/tensor/quantizers/symmetric.py

```diff
@@ -27,7 +29,15 @@ class SymmetricQuantizer(Function):
     """A standard symmetric quantizer."""
 
     @staticmethod
-    def forward(ctx, base: torch.Tensor, qtype: qtype, axis: int, scale: torch.Tensor):
+    def forward(
+        ctx,
+        base: torch.Tensor,
+        qtype: qtype,
+        axis: int,
+        scale: torch.Tensor,
+        activation_qtype: Optional[qtype] = None,
+        tensor_type: Optional[str] = None,
```

Comment:
Passing `activation_qtype` and `tensor_type` is quite ugly here, but as it appears the quantizer needs to initialize the tensor subclass (which requires the knowledge of whether it is a weight or activation quantized, and what is the qtype of the activation), I don't really see any way around?

Reply:
As explained before, the main constraint is to have the call to the constructor embedded in a Function in graphs that dynamically creates QTensor objects. It would be perhaps better to wrap the calls to the quantizers in Function inside each specialized QTensor class.